src/orion/core/io/database/mongodb.py

```diff
@@ -80,7 +80,7 @@ class MongoDB(AbstractDB):
     def __init__(self, host='localhost', name=None,
                  port=None, username=None, password=None):
         """Init method, see attributes of :class:`AbstractDB`."""
-        self.options = {'authSource': name}
+        self.options['authSource'] = name
```

Comment:
You need to init the Parent class before, otherwise `self.options` does not exist yet.